src/database/config.rs

```diff
@@ -0,0 +1,85 @@
+
+use std::collections::{HashSet, HashMap};
+
+use serde_derive::{Serialize, Deserialize};
+use serde::Serialize;
+use serde::Serializer;
+use serde::Deserialize;
+use serde::Deserializer;
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub enum RankingOrdering {
+    Asc,
+    Dsc
+}
+
+impl Serialize for RankingOrdering {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+        where S: Serializer
+    {
+        serializer.serialize_str(match *self {
+            RankingOrdering::Asc => "asc",
+            RankingOrdering::Dsc => "dsc",
+        })
+    }
+}
+
+impl<'de> Deserialize<'de> for RankingOrdering {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+        where D: Deserializer<'de>
+    {
+        let s = String::deserialize(deserializer)?;
+        Ok(match s.as_str() {
+            "asc" => RankingOrdering::Asc,
+            "dsc" => RankingOrdering::Dsc,
+            _ => RankingOrdering::Dsc
+        })
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct AccessToken {
+    pub read_key: String,
+    pub write_key: String,
+    pub admin_key: String,
+}
+
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct Config {
+    pub stop_words: Option<HashSet<String>>,
+    pub ranking_order: Option<Vec<String>>,
+    pub distinct_field: Option<String>,
+    pub ranking_rules: Option<HashMap<String, RankingOrdering>>,
+    pub access_token: Option<AccessToken>,
+}
+
+impl Config {
+    pub(crate) fn default() -> Config {
+        Config {
+            stop_words: None,
+            ranking_order: None,
+            distinct_field: None,
+            ranking_rules: None,
+            access_token: None,
+        }
+    }
```

Comment:
The `#[derive(Default)]` syntax would have produced the same output, please remove this function and replace it by the derived one.